## Create the working directory

```
mkdir -p /cluster/tufts/biocontainers/workshop/Spring2024/rnaseq
```

## Prepare the input samplesheet

### Samplesheet generated by fetchngs

```
/cluster/tufts/biocontainers/workshop/Spring2024/fetchngs/fetchngsOut/samplesheet
ls
```

You should see the below output:

```
id_mappings.csv  multiqc_config.yml  samplesheet.csv
```

We can check the content of `samplesheet.csv`. We can use this `samplesheet.csv` directly for rnaseq anlaysis, but I prefer to create a simpler samplesheet.

```
head -n 3 samplesheet.csv
```

```
"sample","fastq_1","fastq_2","strandedness","run_accession","experiment_accession","sample_accession","secondary_sample_accession","study_accession","secondary_study_accession","submission_accession","run_alias","experiment_alias","sample_alias","study_alias","library_layout","library_selection","library_source","library_strategy","library_name","instrument_model","instrument_platform","base_count","read_count","tax_id","scientific_name","sample_title","experiment_title","study_title","sample_description","fastq_md5","fastq_bytes","fastq_ftp","fastq_galaxy","fastq_aspera"
"SRX1693951","fetchngsOut/fastq/SRX1693951_SRR3362661_1.fastq.gz","fetchngsOut/fastq/SRX1693951_SRR3362661_2.fastq.gz","auto","SRR3362661","SRX1693951","SAMN04639576","SRS1386868","PRJNA318251","SRP073189","SRA409858","GSM2114336_r1","GSM2114336","GSM2114336","GSE80182","PAIRED","cDNA","TRANSCRIPTOMIC","RNA-Seq","","Illumina HiSeq 2500","ILLUMINA","9527214600","31757382","9606","Homo sapiens","A549_GFPkd_1","Illumina HiSeq 2500 sequencing: GSM2114336: A549_GFPkd_1 Homo sapiens RNA-Seq","A TGFbeta-PRMT5-MEP50 Axis Regulates Cancer Cell Invasion through Histone H3 and H4 Arginine Methylation Coupled Transcriptional Activation and Repression","A549_GFPkd_1","d2da6755e6219a091e0a84e9fc63a6fa;ac0e3cd5d59ef55aa3c6d3bb0c31a10c","3720225514;3596832711","ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_2.fastq.gz","ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_2.fastq.gz","fasp.sra.ebi.ac.uk:/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_1.fastq.gz;fasp.sra.ebi.ac.uk:/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_2.fastq.gz"
"SRX1693952","fetchngsOut/fastq/SRX1693952_SRR3362662_1.fastq.gz","fetchngsOut/fastq/SRX1693952_SRR3362662_2.fastq.gz","auto","SRR3362662","SRX1693952","SAMN04639580","SRS1386867","PRJNA318251","SRP073189","SRA409858","GSM2114337_r1","GSM2114337","GSM2114337","GSE80182","PAIRED","cDNA","TRANSCRIPTOMIC","RNA-Seq","","Illumina HiSeq 2500","ILLUMINA","11198386200","37327954","9606","Homo sapiens","A549_GFPkd_2","Illumina HiSeq 2500 sequencing: GSM2114337: A549_GFPkd_2 Homo sapiens RNA-Seq","A TGFbeta-PRMT5-MEP50 Axis Regulates Cancer Cell Invasion through Histone H3 and H4 Arginine Methylation Coupled Transcriptional Activation and Repression","A549_GFPkd_2","152754e5004274d7e0d8f56ea481e2d7;f2290b266571d98c782e7614528ab3c9","4367522067;4230444730","ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_2.fastq.gz","ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_2.fastq.gz","fasp.sra.ebi.ac.uk:/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_1.fastq.gz;fasp.sra.ebi.ac.uk:/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_2.fastq.gz"
```

### Reformating samplesheet

```
awk -F"," 'OFS=","{print $27,$2,$3,$4}' samplesheet.csv | sed 's/A549_//g' | sed 's/sample_title/sample/g'> /cluster/tufts/biocontainers/workshop/Spring2024/rnaseq/samplesheet.csv
```

Using `cat` we can see that our newly generated samplesheet is simple and easy to read.

```
cat /cluster/tufts/biocontainers/workshop/Spring2024/rnaseq/samplesheet.csv
```

```
"sample","fastq_1","fastq_2","strandedness"
"GFPkd_1","fetchngsOut/fastq/SRX1693951_SRR3362661_1.fastq.gz","fetchngsOut/fastq/SRX1693951_SRR3362661_2.fastq.gz","auto"
"GFPkd_2","fetchngsOut/fastq/SRX1693952_SRR3362662_1.fastq.gz","fetchngsOut/fastq/SRX1693952_SRR3362662_2.fastq.gz","auto"
"GFPkd_3","fetchngsOut/fastq/SRX1693953_SRR3362663_1.fastq.gz","fetchngsOut/fastq/SRX1693953_SRR3362663_2.fastq.gz","auto"
"PRMT5kd_1","fetchngsOut/fastq/SRX1693954_SRR3362664_1.fastq.gz","fetchngsOut/fastq/SRX1693954_SRR3362664_2.fastq.gz","auto"
"PRMT5kd_2","fetchngsOut/fastq/SRX1693955_SRR3362665_1.fastq.gz","fetchngsOut/fastq/SRX1693955_SRR3362665_2.fastq.gz","auto"
"PRMT5kd_3","fetchngsOut/fastq/SRX1693956_SRR3362666_1.fastq.gz","fetchngsOut/fastq/SRX1693956_SRR3362666_2.fastq.gz","auto"
"MEP50kd_1","fetchngsOut/fastq/SRX1693957_SRR3362667_1.fastq.gz","fetchngsOut/fastq/SRX1693957_SRR3362667_2.fastq.gz","auto"
"MEP50kd_2","fetchngsOut/fastq/SRX1693958_SRR3362668_1.fastq.gz","fetchngsOut/fastq/SRX1693958_SRR3362668_2.fastq.gz","auto"
"MEP50kd_3","fetchngsOut/fastq/SRX1693959_SRR3362669_1.fastq.gz","fetchngsOut/fastq/SRX1693959_SRR3362669_2.fastq.gz","auto"
```

You may notice that the `fastq_1` and `fastq_2` columns contain the location of fastq files. However, these are **relative paths** instead of **absolute paths**. To make sure the `rnaseq` pipeline can locate the fastq files, we can create a soft link for `fetchngsOut` in the working directory of `rnaseq` pipeline.

```
cd /cluster/tufts/biocontainers/workshop/Spring2024/rnaseq/
ln -s /cluster/tufts/biocontainers/workshop/Spring2024/fetchngs/fetchngsOut .
ls -l
```

You should see that we linked `fetchngsOut` to the current directory. 
```
lrwxrwxrwx 1 yzhang85 biotools   69 Mar  1 14:18 fetchngsOut -> /cluster/tufts/biocontainers/workshop/Spring2024/fetchngs/fetchngsOut/
-rw-rw---- 1 yzhang85 biotools 1163 Mar  1 14:16 samplesheet.csv
```

## rnaseq on Open OnDemand
Using `fetchngs` we already downloaded the raw fastq files for RNAseq. However, to conduct RNAseq analysis we also need the reference genome fasta file and gtf annotation file. Since these are human samples. We will need the human reference genome. 

To obtain human reference genome, there are two options:
1. Choose `GRCh38` in `iGenomes`. This method is simple. I already set up the `iGenomes` for users. You only need to select which reference to use. However, this method is `not recommended`, because the annotation files in `iGenomes` have not been updated in some years. So they are out of date. **Do not use them for your own research**. They are good choices for classroom/workshop.
2. Download the latest version of genomes from public database such as `Ensembl` or `NCBI`.

In this workshop, we will show users how to download your own reference genomes. 

### Arguments
- Number of hours: 12
- Select cpu partition: batch
- Reservation for class, training, workshop: Default
- Version: 3.14.0
- Working Directory: `/cluster/tufts/biocontainers/workshop/Spring2024/rnaseq/`   ## Change this to your own directory
- outdir: rnaseqOut
- input: samplesheet.csv
- multiqc_title: PRMT5-MEP50
- iGenomes: None
- fasta: https://ftp.ensembl.org/pub/release-111/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
- gtf: https://ftp.ensembl.org/pub/release-111/gtf/homo_sapiens/Homo_sapiens.GRCh38.111.gtf.gz
- trimmer: trimgalor
- extra_fastp_args: -q 35 --paired
- aligner: star_salmon
- save_reference: true
- skip_bbsplit: true
- skip_umi_extract: true
- skip_pseudo_alignment: true
- skip_stringtie: true
-  skip_preseq: false

```
------------------------------------------------------
                                        ,--./,-.
        ___     __   __   __   ___     /,-._.--~'
  |\ | |__  __ /  ` /  \ |__) |__         }  {
  | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                        `._,._,'
  nf-core/rnaseq v3.14.0
------------------------------------------------------
Core Nextflow options
  runName                   : spontaneous_thompson
  containerEngine           : singularity
  launchDir                 : /cluster/tufts/biocontainers/workshop/Spring2024/rnaseq
  workDir                   : /cluster/tufts/biocontainers/workshop/Spring2024/rnaseq/work
  projectDir                : /cluster/tufts/biocontainers/nf-core/pipelines/nf-core-rnaseq/3.14.0/3_14_0
  userName                  : yzhang85
  profile                   : tufts
  configFiles               : 

Input/output options
  input                     : samplesheet.csv
  outdir                    : rnaseqOut
  multiqc_title             : PRMT5-MEP50

Reference genome options
  fasta                     : https://ftp.ensembl.org/pub/release-111/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
  gtf                       : https://ftp.ensembl.org/pub/release-111/gtf/homo_sapiens/Homo_sapiens.GRCh38.111.gtf.gz
  igenomes_base             : /cluster/tufts/biocontainers/datasets/igenomes/

Read trimming options
  extra_trimgalore_args     : -q 35 --paired
  min_trimmed_reads         : 50

UMI options
  umitools_grouping_method  : unique

Alignment options
  pseudo_aligner            : salmon
  salmon_quant_libtype      : A

Optional outputs
  save_reference            : true

Quality Control
  deseq2_vst                : false

Process skipping options
  skip_umi_extract          : true
  skip_pseudo_alignment     : true
  skip_stringtie            : true
  skip_preseq               : false

Institutional config options
  config_profile_description: The Tufts University HPC cluster profile provided by nf-core/configs.
  config_profile_contact    : Yucheng Zhang
  config_profile_url        : https://it.tufts.edu/high-performance-computing

Max job request options
  max_cpus                  : 72
  max_memory                : 120 GB
  max_time                  : 7d

!! Only displaying parameters that differ from the pipeline defaults !!
------------------------------------------------------
If you use nf-core/rnaseq for your analysis please cite:

* The pipeline
  https://doi.org/10.5281/zenodo.1400710

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://github.com/nf-core/rnaseq/blob/master/CITATIONS.md

[-        ] process > NFCORE_RNASEQ:RNASEQ:PREPAR... -
[-        ] process > NFCORE_RNASEQ:RNASEQ:PREPAR... -
[-        ] process > NFCORE_RNASEQ:RNASEQ:PREPAR... -
[-        ] process > NFCORE_RNASEQ:RNASEQ:PREPAR... -
[-        ] process > NFCORE_RNASEQ:RNASEQ:PREPAR... -
[-        ] process > NFCORE_RNASEQ:RNASEQ:PREPAR... -
```