## Create the working directory

```
mkdir -p /cluster/tufts/yzhang85/rt/yzhang85/nf-core_workshp/rnaseq
```

## Prepare the input samplesheet

### Samplesheet generated by fetchngs

```
cd /cluster/tufts/yzhang85/rt/yzhang85/nf-core_workshp/fetchngs/fetchngsOut/samplesheet
ls
```

You should see the below output:

```
id_mappings.csv  multiqc_config.yml  samplesheet.csv
```

We can check the content of `samplesheet.csv`. We can use this `samplesheet.csv` directly for rnaseq anlaysis, but I prefer to create a simpler samplesheet.

```
head -n 3 samplesheet.csv
```

```
"sample","fastq_1","fastq_2","strandedness","run_accession","experiment_accession","sample_accession","secondary_sample_accession","study_accession","secondary_study_accession","submission_accession","run_alias","experiment_alias","sample_alias","study_alias","library_layout","library_selection","library_source","library_strategy","library_name","instrument_model","instrument_platform","base_count","read_count","tax_id","scientific_name","sample_title","experiment_title","study_title","sample_description","fastq_md5","fastq_bytes","fastq_ftp","fastq_galaxy","fastq_aspera"
"SRX1693951","fetchngsOut/fastq/SRX1693951_SRR3362661_1.fastq.gz","fetchngsOut/fastq/SRX1693951_SRR3362661_2.fastq.gz","auto","SRR3362661","SRX1693951","SAMN04639576","SRS1386868","PRJNA318251","SRP073189","SRA409858","GSM2114336_r1","GSM2114336","GSM2114336","GSE80182","PAIRED","cDNA","TRANSCRIPTOMIC","RNA-Seq","","Illumina HiSeq 2500","ILLUMINA","9527214600","31757382","9606","Homo sapiens","A549_GFPkd_1","Illumina HiSeq 2500 sequencing: GSM2114336: A549_GFPkd_1 Homo sapiens RNA-Seq","A TGFbeta-PRMT5-MEP50 Axis Regulates Cancer Cell Invasion through Histone H3 and H4 Arginine Methylation Coupled Transcriptional Activation and Repression","A549_GFPkd_1","d2da6755e6219a091e0a84e9fc63a6fa;ac0e3cd5d59ef55aa3c6d3bb0c31a10c","3720225514;3596832711","ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_2.fastq.gz","ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_2.fastq.gz","fasp.sra.ebi.ac.uk:/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_1.fastq.gz;fasp.sra.ebi.ac.uk:/vol1/fastq/SRR336/001/SRR3362661/SRR3362661_2.fastq.gz"
"SRX1693952","fetchngsOut/fastq/SRX1693952_SRR3362662_1.fastq.gz","fetchngsOut/fastq/SRX1693952_SRR3362662_2.fastq.gz","auto","SRR3362662","SRX1693952","SAMN04639580","SRS1386867","PRJNA318251","SRP073189","SRA409858","GSM2114337_r1","GSM2114337","GSM2114337","GSE80182","PAIRED","cDNA","TRANSCRIPTOMIC","RNA-Seq","","Illumina HiSeq 2500","ILLUMINA","11198386200","37327954","9606","Homo sapiens","A549_GFPkd_2","Illumina HiSeq 2500 sequencing: GSM2114337: A549_GFPkd_2 Homo sapiens RNA-Seq","A TGFbeta-PRMT5-MEP50 Axis Regulates Cancer Cell Invasion through Histone H3 and H4 Arginine Methylation Coupled Transcriptional Activation and Repression","A549_GFPkd_2","152754e5004274d7e0d8f56ea481e2d7;f2290b266571d98c782e7614528ab3c9","4367522067;4230444730","ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_2.fastq.gz","ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_2.fastq.gz","fasp.sra.ebi.ac.uk:/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_1.fastq.gz;fasp.sra.ebi.ac.uk:/vol1/fastq/SRR336/002/SRR3362662/SRR3362662_2.fastq.gz"
```

### Reformating samplesheet

```
awk -F"," 'OFS=","{print $27,$2,$3,$4}' samplesheet.csv > /cluster/tufts/yzhang85/rt/yzhang85/nf-core_workshp/rnaseq/samplesheet.csv
```

Using `cat` we can see that our newly generated samplesheet is simple and easy to read.

```
cat /cluster/tufts/yzhang85/rt/yzhang85/nf-core_workshp/rnaseq/samplesheet.csv
```

```
"sample_title","fastq_1","fastq_2","strandedness"
"A549_GFPkd_1","fetchngsOut/fastq/SRX1693951_SRR3362661_1.fastq.gz","fetchngsOut/fastq/SRX1693951_SRR3362661_2.fastq.gz","auto"
"A549_GFPkd_2","fetchngsOut/fastq/SRX1693952_SRR3362662_1.fastq.gz","fetchngsOut/fastq/SRX1693952_SRR3362662_2.fastq.gz","auto"
"A549_GFPkd_3","fetchngsOut/fastq/SRX1693953_SRR3362663_1.fastq.gz","fetchngsOut/fastq/SRX1693953_SRR3362663_2.fastq.gz","auto"
"A549_PRMT5kd_1","fetchngsOut/fastq/SRX1693954_SRR3362664_1.fastq.gz","fetchngsOut/fastq/SRX1693954_SRR3362664_2.fastq.gz","auto"
"A549_PRMT5kd_2","fetchngsOut/fastq/SRX1693955_SRR3362665_1.fastq.gz","fetchngsOut/fastq/SRX1693955_SRR3362665_2.fastq.gz","auto"
"A549_PRMT5kd_3","fetchngsOut/fastq/SRX1693956_SRR3362666_1.fastq.gz","fetchngsOut/fastq/SRX1693956_SRR3362666_2.fastq.gz","auto"
"A549_MEP50kd_1","fetchngsOut/fastq/SRX1693957_SRR3362667_1.fastq.gz","fetchngsOut/fastq/SRX1693957_SRR3362667_2.fastq.gz","auto"
"A549_MEP50kd_2","fetchngsOut/fastq/SRX1693958_SRR3362668_1.fastq.gz","fetchngsOut/fastq/SRX1693958_SRR3362668_2.fastq.gz","auto"
"A549_MEP50kd_3","fetchngsOut/fastq/SRX1693959_SRR3362669_1.fastq.gz","fetchngsOut/fastq/SRX1693959_SRR3362669_2.fastq.gz","auto"
```

You may notice that the `fastq_1` and `fastq_2` columns contain the location of fastq files. However, these are **relative paths** instead of **absolute paths**. To make sure the `rnaseq` pipeline can locate the fastq files, we can create a soft link for `fetchngsOut` in the working directory of `rnaseq` pipeline.

```
cd /cluster/tufts/yzhang85/rt/yzhang85/nf-core_workshp/rnaseq
ln -s /cluster/tufts/yzhang85/rt/yzhang85/nf-core_workshp/fetchngs/fetchngsOut .
```
